src/math/CurveFitting.cc

```diff
@@ -200,41 +174,64 @@ namespace pfs{ namespace drp{ namespace stella{ namespace math{
       if (P_D_A1_XRange->getShape()[0] != 2){
         string message("pfs::drp::stella::math::CurveFitting::PolyFit: ERROR: P_D_A1_XRange->getShape()[0](=");
         message += to_string(P_D_A1_XRange->getShape()[0]) + " != 2";
-        cout << message << endl;
         throw LSST_EXCEPT(pexExcept::Exception, message.c_str());
       }
       xRange.deep() = *P_D_A1_XRange;
-      #ifdef __DEBUG_POLYFIT__
-        cout << "pfs::drp::stella::math::CurveFitting::PolyFit: *P_D_A1_XRange set to " << *P_D_A1_XRange << endl;
-      #endif
+      LOGLS_DEBUG(_log, "*P_D_A1_XRange set to " << *P_D_A1_XRange);
     }
-    ndarray::Array<T, 1, 1> xNew;
+    ndarray::Array<T, 1, 1> xNew = ndarray::allocate(D_A1_X_In.getShape()[0]);
+
     /// shift and rescale x_In to fit into range [-1.,1.]
     if ((std::fabs(xRange[0] + 1.) > 0.00000001) || (std::fabs(xRange[1] - 1.) > 0.00000001)){
       xNew = pfs::drp::stella::math::convertRangeToUnity(D_A1_X_In, xRange);
     }
     else{
       xNew = D_A1_X_In;
     }
-    #ifdef __DEBUG_POLY__
-      cout << "pfs::drp::stella::math::CurveFitting::Poly: D_A1_X_In = " << D_A1_X_In << endl;
-      cout << "pfs::drp::stella::math::CurveFitting::Poly: xNew = " << xNew << endl;
-    #endif
+    LOGLS_DEBUG(_log, "D_A1_X_In = " << D_A1_X_In);
+    LOGLS_DEBUG(_log, "xNew = " << xNew);
+    LOGLS_DEBUG(_log, "xRange = " << xRange);
 
-    std::vector<T> D_A1_X(xNew.begin(), xNew.end());
+    std::vector<T> D_A1_X(D_A1_X_In.begin(), D_A1_X_In.end());
     std::vector<T> D_A1_Y(D_A1_Y_In.begin(), D_A1_Y_In.end());
-    std::vector<T> D_A1_X_New(D_A1_X.size());
-    std::vector<T> D_A1_Y_New(D_A1_Y.size());
-    std::vector<T> D_A1_MeasureErrors(D_A1_X.size());
-    std::vector<T> D_A1_MeasureErrors_New(D_A1_X.size());
     std::vector<size_t> I_A1_OrigPos(D_A1_X_In.getShape()[0]);
+    std::vector<T> V_MeasureErrors(P_D_A1_MeasureErrors->begin(), P_D_A1_MeasureErrors->end());
+    LOGLS_DEBUG(_log, "V_MeasureErrors = " << V_MeasureErrors);
+
+    PTR(ndarray::Array<T, 1, 1>) P_D_A1_MeasureErrorsTemp(new ndarray::Array<T, 1, 1>(ndarray::external(V_MeasureErrors.data(), 
+                                                                                                        ndarray::makeVector(int(V_MeasureErrors.size())), 
+                                                                                                        ndarray::makeVector(1))));
```

Comment:
For all the changes on this commit (077f70ad770d169bc107de72387b5d14675ab678), more comments would be helpful. The commit message provides a useful roadmap to what's happening (thank you!), but a few signposts to make the code easier to follow would be even better.